x-pack/plugins/discover_log_explorer/public/app_updater.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AppUpdater } from '@kbn/core-application-browser';
+import { DEFAULT_APP_CATEGORIES } from '@kbn/core/public';
+import { i18n } from '@kbn/i18n';
+import { BehaviorSubject } from 'rxjs';
+import { LOG_EXPLORER_PROFILE_ID } from '../common/constants';
+
+const logExplorerDeepLink = {
+  id: LOG_EXPLORER_PROFILE_ID,
+  title: i18n.translate('xpack.discoverLogExplorer.deepLink', {
+    defaultMessage: 'Logs Explorer',
+  }),
+  path: `#/p/log-explorer`,
+  category: DEFAULT_APP_CATEGORIES.observability,
+  euiIconType: 'logoObservability',
+};
+
+export const createAppUpdater = () => {
+  return new BehaviorSubject<AppUpdater>((app) => {
+    if (app.id === 'discover') {
+      return {
+        deepLinks: [logExplorerDeepLink],
+      };
+    }
```

Comment:
This is very wrong isolation of control if you ask me.

In theory, the plugin registering the `discover` app should be in charge of doing this (with some kind of IOC).

But even if you don't, **at least** add the `logExplorerDeepLink` to the list of links instead of erasing the existing list. Otherwise, existing deep links will just disappear (e.g if you had two updaters doing that exact same things to add a deep link to an existing app)

Reply:
> This seems a wrong isolation of concerns if you ask me. Global app updaters weren't really designed with that kind of workarounds in mind...

I see how it looks like, we did this to keep the registration of the customization profile and the creation of the deep link isolated into the log-explorer plugin to keep Discover the most agnostic about what customization profiles can be applied. My reasons for registering this here (even if I agree it is not exactly the [use case](https://github.com/elastic/kibana/blob/main/packages/core/application/core-application-browser/src/contracts.ts#L27) for the `registerAppUpdater` function) was answering the question "why does discover register a deep link for a profile it doesn't know if exist or under which conditions is registered?".
We want to keep this profile logic injectable as much as possible to avoid affecting the current discover implementation.
But I'm open to reconsidering this implementation, do you have anything in mind that can help us do this better?

>  we need to at least add the logExplorerDeepLink to the list of links instead of erasing the existing ones.

You right, I didn't think about this scenario as there were no deepLinks registered in discover, I'll update the implementation right now, good catch!